sql/src/main/java/org/apache/druid/sql/calcite/planner/Calcites.java

```diff
@@ -173,6 +173,8 @@ public static ColumnType getValueTypeForRelDataTypeFull(final RelDataType type)
         return ColumnType.LONG_ARRAY;
       }
       return ColumnType.STRING_ARRAY;
+    } else if (sqlTypeName == SqlTypeName.NULL) {
+      return ColumnType.STRING;
```

Comment:
why is this needed?

Reply:
```
org.apache.druid.java.util.common.ISE: Cannot translate sqlTypeName[NULL] to Druid type for field[EXPR$0]
	at org.apache.druid.sql.calcite.table.RowSignatures.fromRelDataType(RowSignatures.java:68)
	at org.apache.druid.sql.calcite.rule.DruidLogicalValuesRule.onMatch(DruidLogicalValuesRule.java:75)
	at org.apache.calcite.plan.volcano.VolcanoRuleCall.onMatch(VolcanoRuleCall.java:208)
```
We want to build the DruidTable object with a valid RowSignature from the inline datasource. RowSignature needs a non-null column type for each column. When there is just row with null value in the inline data, the sql type is null. When we don't have a corresponding druid column type, we throw the above exception. 

Instead of introducing a new column type, I have mapped the null type to string type.

Reply:
Correction - RowSignature allows non-null column type. The exception is thrown during the conversion of signature from calcite type to druid type.